src/runtime_src/doc/toc/p2p.rst

```diff
@@ -151,3 +151,133 @@ PCIe Topology Considerations
+  - Source buffer (buf_src) is OpenCL buffer resident of Card1's DDR
+  - Destination buffer (buf_dst) is OpenCL buffer resident of Card2's DDR
+
+Typical coding style:
+
+  1. In the OpenCL host code, create separate `cl_context` for each `cl_device_id`
+  2. Define `buf_src` as regular buffer
+  3. Define `buf_dst` as P2P buffer
+  4. Import the P2P buffer or buf_dst to the context of buf_src. Use the following APIs
+
+       - `xclGetMemObjectFd`
+       - `xclGetMemObjectFromFd`
+  5. Perform the copy operation from `buf_src` to `imported_dst_buf`
+
+.. code-block:: cpp
+
+   cl_mem src_buf; // Source Buffer (regular) allocated in DDR bank0
+   cl_mem_ext_ptr_t src_buf_ext = {0};
+   src_buf_ext.flags = XCL_MEM_DDR_BANK0;
+   src_buf = clCreateBuffer(src_context, CL_MEM_WRITE_ONLY | CL_MEM_EXT_PTR_XILINX, buffersize, &src_buf_ext, &err);
+
+
+   cl_mem dst_buf;  // Destination buffer (P2P) allocated in DDR bank0 of destination context
+   cl_mem_ext_ptr_t dst_buf_ext = {0};
+
+   dst_buf_ext.flags = XCL_MEM_DDR_BANK0 | XCL_MEM_EXT_P2P_BUFFER;
+   dst_buf = clCreateBuffer(dst_context, CL_MEM_READ_ONLY | CL_MEM_EXT_PTR_XILINX, buffersize, &dst_buf_ext, &err);
+
+   // Import Destination P2P buffer to the source context
+   err = xclGetMemObjectFd(dst_buf, &fd);
+
+   cl_mem imported_dst_buf;
+
+   err = xclGetMemObjectFromFd(src_context, device_id[0], 0, fd, &imported_dst_buf); // Import
+
+   // Copy Operation: Local Source buffer -> Imported Destination Buffer
+
+   err = clEnqueueCopyBuffer(src_command_queue, src_buf, imported_dst_buf, 0, 0, sizeof(data_t)*LENGTH, 0, NULL, &event); 
+
+
+Profile Report
+..............
+
+In the Profile Summary report file the P2P transfer is shown under **Data Transfer: DMA Bypass** 
+
+**Data Transfer: DMA Bypass**
+
++-------+----------------+-----------+------------+-----------+----------+----------+-------------+
+| Device|  Transfer Type | Number of |  Transfer  | Total Data| Total    | Average  | Average     |
+|       |                | Transfer  |  Rate(MB/s)| Transfer  | Time (ms)| Size (Kb)| Latency(ns) |     
++=======+================+===========+============+===========+==========+==========+=============+
+| ...   |     IN         |     4096  |    N/A     |    0.262  |    N/A   |   0.064  |      N/A    |
++-------+----------------+-----------+------------+-----------+----------+----------+-------------+
+
+The report shows the P2P transfer corresponds to the receiving device (i.e. transfer type IN).
+
+
+P2P Data Transfer between FPGA Card and NVMe Device 
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Using the P2P enabled device the data can be transferred between the FPGA device and another NVMe Device, such as SMART SSD, without migrating the data via host memory space.
+
+OpenCL coding style
+...................
+
+Typical coding style 
+
+   1. Create P2P buffer
+   2. Map P2P buffer to the host space
+   3. Access the SSD location through Linux File function
```

Comment:
through linux file system. the file needs to be opened with O_DIRECT.

Reply:
Done, modified the text as suggested